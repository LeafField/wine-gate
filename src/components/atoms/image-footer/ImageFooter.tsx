import React, { FC } from "react";
import { Wines } from "../../../types/article_types";

type Props = Pick<Wines, "author_name" | "price">;

const ImageFooter: FC<Props> = ({ author_name, price }) => {
  return (
    <div className="flex items-center justify-between">
      <p className="text-main-text">{author_name}</p>
      <p className="text-main-text">
        {price}円{"(参考価格)"}
      </p>
    </div>
  );
};

export default ImageFooter;
