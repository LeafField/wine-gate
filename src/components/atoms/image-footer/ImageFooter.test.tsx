import { render, screen } from "@testing-library/react";
import ImageFooter from "./ImageFooter";

describe("ImageFooterの単体テスト", () => {
  it("propsに渡した値通りに正しくレンダリングされているか", () => {
    const { container } = render(
      <ImageFooter author_name="Test Author" price={1000} />,
    );

    expect(screen.getByText("Test Author")).toBeInTheDocument();
    expect(screen.getByText("1000円(参考価格)")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
