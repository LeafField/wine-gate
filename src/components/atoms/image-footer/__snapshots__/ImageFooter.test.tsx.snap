// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageFooterの単体テスト propsに渡した値通りに正しくレンダリングされているか 1`] = `
<div>
  <div
    class="flex items-center justify-between"
  >
    <p
      class="text-main-text"
    >
      Test Author
    </p>
    <p
      class="text-main-text"
    >
      1000
      円
      (参考価格)
    </p>
  </div>
</div>
`;
